dl/src/main/scala/com/intel/analytics/bigdl/dataset/text/SampleToBatchPadding.scala

```diff
@@ -0,0 +1,184 @@
+      private var featureData: Array[T] = null
+      private var labelData: Array[T] = null
+      private var sampleData: Array[Sample[T]] = null
+
+      private val batchSize = Utils.getBatchSize(totalBatch)
+      private var featureSize: Array[Int] = null
+      private var labelSize: Array[Int] = null
+      private var oneFeatureLength: Int = 0
+      private var oneLabelLength: Int = 0
+      private var padFeatureLength: Int = 0
+      private var padLabelLength: Int = 0
+      override def hasNext: Boolean = prev.hasNext
+
+      override def next(): MiniBatch[T] = {
+        if (prev.hasNext) {
+          var i = 0
+          var maxfeatureElement = 0
+          var maxlabelElement = 0
+          var labelIndex = 0
+          var featureIndex = 0
+          var padLength: Int = 0
+          if (sampleData == null) sampleData = new Array[Sample[T]](batchSize)
+          while (i < batchSize && prev.hasNext) {
+            val sample = prev.next()
+            require(sample.feature().isContiguous() && sample.label().isContiguous(),
+              "SampleToBatchPadding: Only support contiguous tensor, pls use " +
+                "tensor.contiguous() before batching")
+            if (sampleData(i) == null) sampleData(i) = Sample()
+            if ((maxfeatureElement != 0) && (!padFeature)) {
+              require(sample.feature().nElement() == maxfeatureElement,
+                "SampleToBatchPadding: all feature size should be same when not padding feature")
+            }
+            if ((maxlabelElement != 0) && (!padLabel)) {
+              require(sample.label().nElement() == maxlabelElement,
+                "SampleToBatchPadding: all label size should be same when not padding label")
+            }
+            sampleData(i).copy(sample)
+            if (sample.feature().nElement() > maxfeatureElement) {
+              maxfeatureElement = sample.feature().nElement()
+              featureIndex = i
+            }
+            if (sample.label().nElement() > maxlabelElement) {
+              maxlabelElement = sample.label().nElement()
+              labelIndex = i
+            }
+            i += 1
+          }
+          val batchLength = i
+          if (featureSize == null) {
+            featureSize = Array(1) ++ sampleData(featureIndex).feature().size()
+            labelSize = Array(1) ++ sampleData(labelIndex).label().size()
+          }
+          // init
+          padFeatureLength = sampleData(featureIndex).feature().size(padFeatureDim)
+          padLength = if (padFeature) fixLength.getOrElse(padFeatureLength) else padFeatureLength
+          require(padLength >= padFeatureLength, "SampleToBatchPadding: fixLength " +
+            "should not be less than first dimension of feature when padding")
+          featureSize(padFeatureDim) = padLength
+          featureSize(0) = 1
+          oneFeatureLength = featureSize.product
+          featureSize(0) = batchLength
+
```

Comment:
write a private function both for feature and label?

private def getSize(size: Array[Int], length: Int, fixLength: Option[Int], isPad: Boolean): Unit = {
}